mycroft/skills/core.py

```diff
@@ -591,45 +590,21 @@ def wrapper(message):
 
                 stopwatch = Stopwatch()
                 with stopwatch:
-                    if need_self:
-                        # When registring from decorator self is required
-                        if len(getargspec(handler).args) == 2:
-                            handler(self, unmunge_message(message,
-                                                          self.skill_id))
-                        elif len(getargspec(handler).args) == 1:
-                            handler(self)
-                        elif len(getargspec(handler).args) == 0:
-                            # Zero may indicate multiple decorators, trying the
-                            # usual call signatures
-                            try:
-                                handler(self, unmunge_message(message,
-                                                              self.skill_id))
-                            except TypeError:
-                                handler(self)
-                        else:
-                            LOG.error("Unexpected argument count:" +
-                                      str(len(getargspec(handler).args)))
-                            raise TypeError
+                    is_bound = bool(getattr(handler, 'im_self', None))
+                    num_args = len(getargspec(handler).args) - is_bound
+                    if num_args == 0:
+                        handler()
                     else:
-                        if len(getargspec(handler).args) == 2:
-                            handler(unmunge_message(message, self.skill_id))
-                        elif len(getargspec(handler).args) == 1:
-                            handler()
-                        else:
-                            LOG.error("Unexpected argument count:" +
-                                      str(len(getargspec(handler).args)))
-                            raise TypeError
+                        handler(message)
                     self.settings.store()  # Store settings if they've changed
 
             except Exception as e:
                 # Convert "MyFancySkill" to "My Fancy Skill" for speaking
-                handler_name = re.sub("([a-z])([A-Z])", "\g<1> \g<2>",
-                                      self.name)
-                # TODO: Localize
-                self.speak("An error occurred while processing a request in " +
-                           handler_name)
-                LOG.error("An error occurred while processing a request in " +
-                          self.name, exc_info=True)
+                handler_name = re.sub(r"([a-z])([A-Z])", r"\1 \2", self.name)
+                data = {'skill': handler_name}
```

Comment:
This overwrites the data prepared previously for the skill.handler.started / .stopped messages (the `name` key will be slightly different)

Reply:
Nice catch! Fixed.